taca/analysis/analysis_element.py

```diff
@@ -0,0 +1,163 @@
+"""Analysis methods for sequencing runs produced by Element instruments."""
+
+import glob
+import logging
+import os
+
+from taca.element.Aviti_Runs import Aviti_Run
+from taca.utils.config import CONFIG
+from taca.utils.misc import send_mail
+
+logger = logging.getLogger(__name__)
+
+
+def run_preprocessing(given_run):
+    """Run demultiplexing in all data directories.
+
+    :param str given_run: Process a particular run instead of looking for runs
+    """
+
+    def _process(run):
+        """Process a run/flowcell and transfer to analysis server.
+
+        :param taca.element.Run run: Run to be processed and transferred
+        """
+        try:
+            run.parse_run_parameters()
+        except FileNotFoundError:
+            logger.warning(
+                f"Cannot reliably set NGI_run_id for {run} due to missing RunParameters.json. Aborting run processing"
+            )
+            email_subject = f"Issues processing {run}"
+            email_message = (
+                f"RunParameters.json missing for {run}. Processing was aborted."
```

Comment:
Would this be triggered by some runs right in the beginning of sequencing? If the directory has been created but not yet the RunParameters.json? I guess then it's a bit noisy to send emails. I think maybe it's best to just wait and see how it turns out in reality though

Reply:
I agree it's intuitively a bit noisy, but if we want to wait instead then wouldn't we need a new functionality to discern how long to wait for? Do we know if there is a significant time frame in which we have a run dir but no .json?

Reply:
No i guess we don't really know. So let's wait and see if it's a problem or not